### PR TITLE
PID calibration

### PR DESCRIPTION
**Pull request by AidanSmith**
I am using the Ziegler-Nichols PID control method. It seems to work fine on the practice robot. I have included values that work for the practice robot. We will have to recalibrate for Helios. Ultimate gain is what value of P the robot oscillates, and oscillation period is how long it takes.

### CODE REVIEW

File: QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java [hunk 1]
```diff
@@ -144,9 +144,17 @@ public static void setup() {
...
         recalibrateButton.send(calibrate);
 
-        FloatStatus p = ControlInterface.mainTuning.getFloat("main-drive-p", 0.01f);
-        FloatStatus i = ControlInterface.mainTuning.getFloat("main-drive-i", 0f);
-        FloatStatus d = ControlInterface.mainTuning.getFloat("main-drive-d", 0f);
+        FloatStatus ultgain = ControlInterface.mainTuning.getFloat("drive-PID-ultimate-gain", .0162f);
+        FloatStatus period = ControlInterface.mainTuning.getFloat("drive-PID-oscillation-period", 2f);
+        BooleanStatus calibrating = ControlInterface.mainTuning.getBoolean("calibrating-drive-PID", false);
+
+        FloatInput p = Mixing.select(calibrating, FloatMixing.multiplication.of((FloatInput) ultgain, .6f), ultgain);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java]: 0.6f sounds like a magic number. Either it should be a named constant or should be tunable.

File: QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java [hunk 2]
```diff
@@ -144,9 +144,17 @@ public static void setup() {
...
+        FloatInput i = Mixing.select(calibrating, FloatMixing.division.of(FloatMixing.multiplication.of(p, 2f), (FloatInput) period), FloatMixing.always(0));
+        FloatInput d = Mixing.select(calibrating, FloatMixing.division.of(FloatMixing.multiplication.of(p, (FloatInput) period), 8f), FloatMixing.always(0));
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java]: Same with 2f and 8f.

**Comment by AidanSmith** [comment]
I changed the constants to be tunable.

**Comment by celskeggs** [comment]
Looks good.

**Pull request merged**